src/test/AludelV3.t.sol

```diff
@@ -0,0 +1,308 @@
+// SPDX-License-Identifier: GPL-3.0-or-later
+// solhint-disable func-name-mixedcase
+pragma solidity ^0.8.6;
+
+import {DSTest} from "ds-test/src/test.sol";
+import {ERC20} from "solmate/src/tokens/ERC20.sol";
+import {Vm} from "forge-std/src/Vm.sol";
+import {Ownable} from "@openzeppelin/contracts/access/Ownable.sol";
+
+import {AludelFactory} from "../contracts/AludelFactory.sol";
+import {IAludelV3} from "../contracts/aludel/IAludelV3.sol";
+import {IAludel} from "../contracts/aludel/IAludel.sol";
+import {AludelV3} from "../contracts/aludel/AludelV3.sol";
+import {AludelV3Lib} from "../contracts/aludel/AludelV3Lib.sol";
+
+import {RewardPoolFactory} from "alchemist/contracts/aludel/RewardPoolFactory.sol";
+import {PowerSwitchFactory} from "../contracts/powerSwitch/PowerSwitchFactory.sol";
+
+import {MockERC20} from "../contracts/mocks/MockERC20.sol";
+import {Spy} from "./Spy.sol";
+
+import {Crucible, IUniversalVault} from "alchemist/contracts/crucible/Crucible.sol";
+import {CrucibleFactory} from "alchemist/contracts/crucible/CrucibleFactory.sol";
+
+import {User} from "./User.sol";
+import {Utils} from "./Utils.sol";
+import {UserFactory} from "./UserFactory.sol";
+
+import "forge-std/src/console2.sol";
+
+contract AludelV3Test is DSTest {
+    
+    AludelFactory private factory;
+    Vm private vm;
+    AludelV3 private aludel;
+    Spy private spyTemplate;
+
+    User private user;
+    User private anotherUser;
+    User private admin;
+    User private recipient;
+
+    Crucible private crucibleA;
```

Comment:
I won't review this file too much 'cause there's a lot of copypasted code, will wait for a more definitive version instead